android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -103,29 +103,47 @@ open class PolyOut(
 
     open suspend fun readdir(
         id: String
-    ): Array<String> {
-
+    ): Array<Map<String, String>> {
         val fs = Preferences.getFileSystem(context)
+        val retList = mutableListOf(mapOf<String,String>())
+        
+        retList.removeFirst()
```

Comment:
Isn't `retList` guaranteed to be empty at this point?

Reply:
for some reason it was being instantiated as `[ { } ] `and the `push` later on would start from the first index rather than the zeroth

Reply:
Ah, now I see it, you do explicitly put an empty map inside the list as its first element, I missed that. You wanna do something like this:

`val retList = mutableListOf<Map<String, String>>()`